src/it/scala/deduplication/DeduplicationSpec.scala

```diff
@@ -6,172 +6,179 @@ import java.util.concurrent.TimeoutException
 
 import scala.concurrent.ExecutionContext
 import scala.concurrent.duration._
-import scala.collection.JavaConverters._
 
 import cats.effect._
 import cats.effect.concurrent._
 import cats.implicits._
 
-import org.scalatest.matchers.should.Matchers
-import org.scalatest.flatspec.AnyFlatSpec
-import org.scalatestplus.scalacheck.ScalaCheckDrivenPropertyChecks
+import munit._
 
-import com.amazonaws.services.dynamodbv2.AmazonDynamoDBAsync
-import com.amazonaws.services.dynamodbv2.AmazonDynamoDBAsyncClientBuilder
+import software.amazon.awssdk.services.dynamodb.{model => _, _}
 
 import model._
 import Config._
+import dynamodb.DynamoDbProcessRepo
+import dynamodb.DynamoDbConfig
+import org.scalacheck.Arbitrary
 
-
-class DeduplicationSpec extends AnyFlatSpec with Matchers with ScalaCheckDrivenPropertyChecks {
+class DeduplicationSpec extends FunSuite {
 
   implicit val ec = ExecutionContext.global
   implicit val contextShift = IO.contextShift(ec)
   implicit val timer: Timer[IO] = IO.timer(ec)
 
-  behavior of "Deduplication"
-
-  it should "always process event for the first time" in forAll {
-    (processorId: UUID, id1: UUID, id2: UUID) =>
-
-      val result = deduplicationResource(processorId)
-        .use { ps =>
-          for {
-            a <- ps.protect(id1, IO("nonProcessed"), IO("processed"))
-            b <- ps.protect(id2, IO("nonProcessed"), IO("processed"))
-          } yield (a, b)
+  override def munitValueTransforms = super.munitValueTransforms ++ List(
+    new ValueTransform("IO", {
+      case io: IO[_] => io.unsafeToFuture()
+    })
+  )
+
+  def given[A: Arbitrary]: A = Arbitrary.arbitrary[A].sample.get
+
+  test("Deduplication should always process event for the first time") {
```

Comment:
this is not the same as `forAll` though, `forAll` would run the test multiple times across a range of random values. This only run it once, randomly, so can be flaky

Reply:
Yes, but they are all UUID so should it make any difference if you run once, twice or 10 times?

Reply:
haha, ok, I didn't noticed that it is only for UUID